src/app/clusters/basic-information/CodegenIntegration.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/basic-information/BasicInformationCluster.h>
+#include <app/static-cluster-config/BasicInformation.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+#include <lib/support/BitFlags.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::BasicInformation::Attributes;
+using namespace chip::app::Clusters::BasicInformation::StaticApplicationConfig;
+using chip::Protocols::InteractionModel::Status;
+
+namespace {
+
+// BasicInformationCluster implementation is specifically implemented
+// only for the root endpoint (endpoint 0)
+// So either:
+//   - we have a fixed config and it is endpoint 0 OR
+//   - we have a fully dynamic config
+
+static constexpr size_t kBasicInformationFixedClusterCount = BasicInformation::StaticApplicationConfig::kFixedClusterConfig.size();
+
+static_assert((kBasicInformationFixedClusterCount == 0) ||
+                  ((kBasicInformationFixedClusterCount == 1) &&
+                   BasicInformation::StaticApplicationConfig::kFixedClusterConfig[0].endpointNumber == kRootEndpointId),
+              "Basic Information cluster MUST be on endpoint 0");
+
+ServerClusterRegistration gRegistration(BasicInformationCluster::Instance());
+
+} // namespace
+
+void emberAfBasicInformationClusterInitCallback(EndpointId endpointId)
```

Comment:
This is the wrong callback.  See https://github.com/project-chip/connectedhomeip/pull/39698#discussion_r2264527455